internal/initwd/from_module_test.go

```diff
@@ -52,17 +52,17 @@ func TestDirFromModule_registry(t *testing.T) {
 			Name:       "Install",
 			ModuleAddr: "root",
 			Version:    v,
-			LocalPath:  ".terraform/modules/root/hashicorp-terraform-aws-module-installer-acctest-5e87aff",
+			LocalPath:  filepath.Join(dir, ".terraform/modules/root/hashicorp-terraform-aws-module-installer-acctest-5e87aff"),
```

Comment:
I think these were intentionally relative to give a more reasonably-sized string in the console output when installing, but we could get that back by normalizing the path at the UI layer where it gets printed out:

https://github.com/hashicorp/terraform/blob/36bb9b462c3ef67bf05b3c8a29945c7249a346e9/command/hook_module_install.go#L29

Our `normalizePath` function is currently a method of `command.Meta` and so would be hard to call from that spot, but it doesn't look like `normalizePath` actually needs anything else from `command.Meta`, so I wonder if we could just turn it into a top-level function and make `m.normalizePath` be just a thin wrapper around it (so we don't need to go change all the existing callers, etc).

Reply:
I think this is only an issue for these tests, which are passing around the absolute path and using locally-defined hooks:

https://github.com/hashicorp/terraform/blob/master/internal/initwd/module_install_test.go#L368


Here is some debug output from a "real" module:
```
2019/03/11 13:08:35 [TRACE] ModuleInstaller: installing child modules for . into .terraform/modules
2019/03/11 13:08:35 [DEBUG] Module installer: begin pets
2019/03/11 13:08:35 [TRACE] ModuleInstaller: pets is not yet installed
2019/03/11 13:08:35 [TRACE] ModuleInstaller: cleaning directory .terraform/modules/pets prior to install of pets
2019/03/11 13:08:35 [TRACE] ModuleInstaller: pets has local path "./modules"
2019/03/11 13:08:35 [TRACE] ModuleInstaller: pets uses directory from parent: modules
2019/03/11 13:08:35 [DEBUG] Module installer: pets installed at modules
```

Does that make sense, @apparentlymart?

Reply:
I must admit I'm not really sure anymore how these different bits hang together, but if the real, user-facing output from `terraform init` has relative paths then I'm happy. :grinning: